presto-native-execution/presto_cpp/main/http/tests/HttpTest.cpp

```diff
@@ -234,6 +258,133 @@ TEST_P(HttpTestSuite, timedOutRequests) {
   wrapper.stop();
 }
 
+TEST_P(HttpTestSuite, httpConnectTimeout) {
+  const bool useHttps = GetParam();
+  if (!useHttps) {
+    // Cannot test connect timeout for HTTP as HTTP connection establishment is
+    // handled in kernel.
+    return;
+  }
+  auto memoryPool = defaultMemoryManager().addLeafPool("httpTimeouts");
+
+  auto server = getServer(useHttps, 1);
+
+  auto request1 = std::make_shared<AsyncMsgRequestState>();
+  auto request2 = std::make_shared<AsyncMsgRequestState>();
+
+  server->registerGet("/async/msg1", asyncMsg(request1));
+  server->registerGet("/async/msg2", asyncMsg(request2));
+
+  HttpServerWrapper wrapper(std::move(server));
+  auto serverAddress = wrapper.start().get();
+
+  HttpClientFactory clientFactory;
+  auto client1 = clientFactory.newClient(
+      serverAddress,
+      std::chrono::milliseconds(5'000),
+      std::chrono::milliseconds(0),
+      useHttps,
+      memoryPool);
+  auto client2 = clientFactory.newClient(
+      serverAddress,
+      std::chrono::milliseconds(7'000),
+      std::chrono::milliseconds(2'000),
+      useHttps,
+      memoryPool);
+  request1->customFunc = []() {
+    // Holds the thread for 3s to fully occupy the server io pool (size 1), so
+    // that the second request shall wait in io pool queue for at least 3s
+    std::this_thread::sleep_for(std::chrono::milliseconds(3'000));
+  };
+
+  request1->maxWaitMillis = 10;
+  request2->maxWaitMillis = 10;
+
+  auto promisePair1 = folly::makePromiseContract<bool>();
+  request1->requestPromise = std::move(promisePair1.first);
+  auto promisePair3 = folly::makePromiseContract<bool>();
+  request2->requestPromise = std::move(promisePair3.first);
+
+  std::thread thread1([&]() {
```

Comment:
We do run the second http request in the same thread?

Reply:
thread1 and thread2? Not sure I get your question.

Reply:
Sorry, I meant can we run the two requests in the same thread? thanks!

Reply:
Yeah we can but then we will not be able to validate the success of the first request as the second throws. 2 threads might serve better for illustration purpose?